src/main/scala/zio/ftp/Ftp.scala

```diff
@@ -0,0 +1,153 @@
+package zio.ftp
+
+import java.io.{ IOException, InputStream }
+import java.nio.file.attribute.PosixFilePermission
+
+import org.apache.commons.net.ftp.{ FTP, FTPFile, FTPFileFilter, FTPClient => JFTPClient, FTPSClient => JFTPSClient }
+import zio.ftp.settings.FtpSettings
+import zio.stream.{ Stream, ZStream, ZStreamChunk }
+import zio.{ Chunk, Managed, ZIO }
+
+object Ftp {
+
+  trait FtpClient {
+    private[ftp] val client: JFTPClient
+
+    def mlistFile(path: String): FTPFile = client.mlistFile(path)
+
+    def retrieveFileStream(path: String): InputStream = client.retrieveFileStream(path)
+
+    def deleteFile(path: String): Boolean = client.deleteFile(path)
+
+    def removeDirectory(path: String): Boolean = client.removeDirectory(path)
+
+    def makeDirectory(path: String): Boolean = client.makeDirectory(path)
+
+    def listFiles(path: String, filter: FTPFileFilter): List[FTPFile] = client.listFiles(path, filter).toList
+
+    def storeFile(path: String, is: InputStream): Boolean = client.storeFile(path, is)
+
+    private[ftp] def logout(): Unit = client.logout()
+
+    private[ftp] def disconnect(): Unit = client.disconnect()
+  }
+
+  def connect(settings: FtpSettings): Managed[IOException, FtpClient] =
+    Managed.make(
+      taskIO {
+        val ftpClient = if (settings.secure) new JFTPSClient() else new JFTPClient()
+        settings.proxy.foreach(ftpClient.setProxy)
+        ftpClient.connect(settings.host, settings.port)
+        settings.configureConnection(ftpClient)
+
+        val success = ftpClient.login(settings.credentials.username, settings.credentials.password)
+
+        if (settings.binary) {
+          ftpClient.setFileType(FTP.BINARY_FILE_TYPE)
+        }
+
+        if (settings.passiveMode) {
+          ftpClient.enterLocalPassiveMode()
+        }
+        success -> new FtpClient {
+          val client = ftpClient
+        }
+      }.filterOrDie(_._1)(new IOException(s"Fail to connect to server ${settings.host}:${settings.port}"))
```

Comment:
`orDie` should be used sparingly, best to fail with a specific typed error in most cases.